seqr/views/utils/orm_to_json_utils.py

```diff
@@ -693,11 +697,20 @@ def get_project_collaborators_by_username(project, include_permissions=True):
             collaborator, include_permissions, can_edit=True
         )
 
+    if project_has_anvil(project):
+        permission_levels = get_collaborator_permission_levels(user, project.workspace_namespace, project.workspace_name)
+        collaborators.update({
+            collab.username: _get_collaborator_json(collab, include_permissions,
+                                                    can_edit = permission_levels[collab.email] == CAN_EDIT,
+                                                    is_anvil = True)
+            for collab in User.objects.filter(email__in = permission_levels.keys())
+        })
```

Comment:
Sorry for not catching this earlier, but I just realized that if a user has access to a seqr project through AnVIL but has never logged in they won't show up, right? I think from the user's perspective, they want to know everyone who could access the project, regardless of whether or not they have yet (currently, for example, we show collaborators who have never logged in). You could add something here to force them to show up by just creating a json record for those email addresses:
```
for email, permission_level in in  permission_levels.items():
   if email not in collaborators:
      collaborators[email] = {
         'username': email, // just to ensure everything has a unique ID
         'email': email, 
         'isAnvil': True, '
         'hasViewPermissions': True, 
         'hasEditPermissions': permission_level == CAN_EDIT
      }
```

Reply:
Yes. The original code only shows those AnVIL users who have logged in with Google at least once.
I agree that we'd better show all the users who have access to the project no matter they have logged in seqr locally or with AnVIL.
If we do as you've proposed, only those users who have never logged in seqr will be shown as `AnVIL` users. Is that what you want? I think we should show all users who can obtain permissions from AnVIL as `AnVIL` users.

Reply:
We want all users who are in the workspace ACL to be shown as anvil users, regardless of if they've ever logged in. I realize that I got confused about how  the collaborators dictionary was keyed, so we should instead do something like 
```
if project_has_anvil(project):
        permission_levels = get_collaborator_permission_levels(user, project.workspace_namespace, project.workspace_name)
        users_by_email = {u.email: u for u in User.objects.filter(email__in = permission_levels.keys())}
        collaborators.update({
            collab.username: _get_collaborator_json(collab, include_permissions,
                                                    can_edit = permission_levels[collab.email] == CAN_EDIT,
                                                    is_anvil = True)
            for collab in users_by_email.values())
        })
        no_user_emails = set(permission_levels.keys()) - set(users_by_email.keys())
        for email in no_user_emails:
                collaborators[email] = {
                      'username': email, // just to ensure everything has a unique ID
                       'email': email, 
                       'isAnvil': True, '
                       'hasViewPermissions': True, 
                       'hasEditPermissions': permission_levels[email] == CAN_EDIT
                }
```

Reply:
I see what you mean. My concern is that when a project manager wants to change a Google signed-in collaborator's local permissions, the manager has to firstly revoke the collaborator's AnVIL workspace access and then edit the collaborator locally.

Reply:
theres really no point in updating local permissions for users who have access through anvil, because the anvil permissions override the local ones anyways. I think its an edge case thats okay to ignore

Reply:
I found `lastName` and `displayName` fields are required. To avoid further unexpected missing fields, I added all the fields as below:
```
                collaborators[email] = {
                    'username': email,  # to ensure everything has a unique ID
                    'email': email,
                    'isAnvil': True,
                    'hasViewPermissions': True,
                    'hasEditPermissions': permission == CAN_EDIT,
                    'displayName': email,
                    'lastName': email,
                    'is_staff': False,
                    'is_active': True,
                    'first_name': '', 'last_login': '', 'date_joined': '', 'id': '',
                }
```